llvm/include/llvm/IR/IntrinsicsNVVM.td

```diff
@@ -1624,8 +1624,9 @@ def int_nvvm_compiler_error :
 def int_nvvm_compiler_warn :
     Intrinsic<[], [llvm_anyptr_ty], [], "llvm.nvvm.compiler.warn">;
 
-def int_nvvm_reflect :
-  Intrinsic<[llvm_i32_ty], [llvm_anyptr_ty], [IntrNoMem], "llvm.nvvm.reflect">;
+def int_nvvm_reflect : 
+  Intrinsic<[llvm_i32_ty], [llvm_ptr_ty], [IntrNoMem], "llvm.nvvm.reflect">,
+  ClangBuiltin<"__nvvm_reflect">;
```

Comment:
I vaguely recall that OpenCL folks had to use it with a slightly different signature. I think their pointer argument was in an unusual address space, where OCL keeps their string constants. It would be great to double check that the new builtin does not break them.

Reply:
It seems this broke the NVPTX tests as well. I'll try to check w/ OpenCL, but I know next to nothing about the OpenCL NVPTX support, I was surprised when I leanred that `nxptx64-nvidia-` can be anything other than CUDA.

Reply:
Seems to work fine w/ OpenCL, added a test for it.